src/SimpleSAML/Metadata/SAMLBuilder.php

```diff
@@ -123,28 +122,6 @@ public function getEntityDescriptorText(bool $formatted = true): string
     }
 
 
-    /**
-     * Add a SecurityTokenServiceType for ADFS metadata.
-     *
-     * @param array $metadata The metadata with the information about the SecurityTokenServiceType.
-     */
-    public function addSecurityTokenServiceType(array $metadata): void
```

Comment:
I think there may still be a call to this here
https://github.com/simplesamlphp/simplesamlphp/blob/2509a7abc3e7dc54d1dc02d4d9028732cd15b8fb/modules/admin/src/Controller/Federation.php#L287

Reply:
Ah, good catch! I think that entire if-block for adfs-entities can also go now.
If we want to display them in the admin module, the adfs-module should generate them and provide them using a hook

Reply:
I resurrected my ADFS configuration a little bit. I am still not convinced that the setup I have would work in production. 

The I comment out the `addSecurityTokenServiceType`  in Federation I can see the federation page again. Though there is a reasonably blank "In SAML 2.0 Metadata XML format:" in the web page.